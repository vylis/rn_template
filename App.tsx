/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React, { useState } from "react";
import { SafeAreaView, StyleSheet, Text, View, TouchableOpacity } from "react-native";

const App = () => {
  const [show, setShow] = useState<boolean>(false);

  return (
    <SafeAreaView>
      <View testID="view" style={styles.container}>
        <Text testID="title" style={styles.title}>
          Title
        </Text>
        <Text testID="text" style={styles.text}>
          Test App
        </Text>

        <TouchableOpacity
          testID="pressable"
          style={styles.pressable}
          onPress={() => {
            setShow(!show);
          }}
        >
          <Text style={styles.text}> Click Me</Text>
        </TouchableOpacity>
        {show && (
          <Text testID="hidden" style={styles.text}>
            Hidden Word
          </Text>
        )}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 100,
  },
  title: {
    margin: 20,
    fontSize: 24,
    textAlign: "center",
    fontWeight: "600",
  },
  text: {
    margin: 20,
    fontSize: 18,
    textAlign: "center",
    fontWeight: "400",
  },
  pressable: {
    padding: 40,
    backgroundColor: "red",
  },
});

export default App;
