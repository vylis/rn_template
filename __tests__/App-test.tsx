/**
 * @format
 */

import "react-native";
import React from "react";
import App from "../App";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  renderer.create(<App />);
});

test("should render app", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
