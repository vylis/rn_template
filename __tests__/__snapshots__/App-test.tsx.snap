// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "marginTop": 100,
      }
    }
    testID="view"
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "margin": 20,
          "textAlign": "center",
        }
      }
      testID="title"
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "400",
          "margin": 20,
          "textAlign": "center",
        }
      }
      testID="text"
    >
      Test App
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "opacity": 1,
          "padding": 40,
        }
      }
      testID="pressable"
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "400",
            "margin": 20,
            "textAlign": "center",
          }
        }
      >
         Click Me
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
